bundles/statistics/statsgrid2016/components/Legend.js

```diff
@@ -163,15 +157,16 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.Legend', function (sandbox, loc
      * Triggers a new render when needed (if render was called before previous was finished)
      */
     _renderDone: function () {
+        var me = this;
+        if (me._renderQueue.length) {
+            let render = me._renderQueue.shift();
```

Comment:
Do we need to render all requests in the queue or would just the most recent one be enough?

Reply:
Good point, the most recent one probably is enough. I'll test it out.

Reply:
Turns out all the requests in the queue have to be rendered. Otherwise there might be some indicators missing from the legend.